example/spec/parallel/SyncGranuleTask/SyncZeroByteFileSpec.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+const get = require('lodash/get');
+const pAll = require('p-all');
+const pTimeout = require('p-timeout');
+const { pullStepFunctionEvent } = require('@cumulus/message/StepFunctions');
+const { randomId } = require('@cumulus/common/test-utils');
+
+const { createCollection } = require('@cumulus/integration-tests/Collections');
+const { createProvider } = require('@cumulus/integration-tests/Providers');
+
+const { deleteCollection } = require('@cumulus/api-client/collections');
+const { deleteProvider } = require('@cumulus/api-client/providers');
+const { lambda } = require('@cumulus/aws-client/services');
+const S3 = require('@cumulus/aws-client/S3');
+
+const { loadConfig } = require('../../helpers/testUtils');
+const { fetchFakeS3ProviderBuckets } = require('../../helpers/Providers');
+
+describe('The SyncGranule task with a 0 byte file to be synced', () => {
```

Comment:
An integration test is necessary because unit tests with Localstack actually can't reproduce the issue, most likely because Localstack doesn't actually replicate S3's behavior for handling multipart uploads